internal/slicer/slicer.go

```diff
@@ -24,44 +25,54 @@ type RunOptions struct {
 	TargetDir string
 }
 
-func Run(options *RunOptions) (*Report, error) {
+type pathData struct {
+	until   setup.PathUntil
+	mutable bool
+}
 
-	archives := make(map[string]archive.Archive)
-	extract := make(map[string]map[string][]deb.ExtractInfo)
-	pathInfos := make(map[string]setup.PathInfo)
-	report := NewReport(options.TargetDir)
+type contentChecker struct {
+	knownPaths map[string]pathData
+}
 
-	knownPaths := make(map[string]bool)
-	knownPaths["/"] = true
+func (cc *contentChecker) checkMutable(path string) error {
+	if !cc.knownPaths[path].mutable {
+		return fmt.Errorf("cannot write file which is not mutable: %s", path)
+	}
+	return nil
+}
 
-	addKnownPath := func(path string) {
-		if path[0] != '/' {
-			panic("bug: tried to add relative path to known paths")
-		}
-		cleanPath := filepath.Clean(path)
-		slashPath := cleanPath
-		if path[len(path)-1] == '/' && cleanPath != "/" {
-			slashPath += "/"
-		}
-		for {
-			if _, ok := knownPaths[slashPath]; ok {
-				break
```

Comment:
[Comment for reviewer]: In the new function I have removed this checked intentionally. I think that now that `knownPaths` contains more information that just whether a path is known it could happen that we do:
```go
knownPaths[path] = pathData{mutable: true}
addParents(knownPaths, path)
```
and if we kept the code highlighted it would mean that the parents will not be added which will lead to subtle bugs. I think it is better to "duplicate" the work than to rely on the fact that the parents have to always be added before the path.

Reply:
This is no longer the case, the logic has been kept intact.